fetch/mesh_data.go

```diff
@@ -240,6 +261,21 @@ func (f *Fetch) GetLayerOpinions(ctx context.Context, peer p2p.Peer, lid types.L
 	return f.servers[OpnProtocol].Request(ctx, peer, reqData)
 }
 
+func (f *Fetch) peerEpochInfoStreamed(ctx context.Context, peer p2p.Peer, epochBytes []byte) (*EpochData, error) {
+	var ed EpochData
+	if err := f.servers[atxProtocol].StreamRequest(
+		ctx, peer, epochBytes,
+		func(ctx context.Context, s io.ReadWriter) error {
```

Comment:
i don't undersrtand why this interface needs to be so complex. can it just accept a scale.Decodable, as is:

```
var epochInfo EpochInfo
StreamRequest(ctx, peer, epochBytes, &epochInfo)

Reply:
My idea was not to use `EpochInfo` at all at some point, and use a channel of ATX IDs instead. But I'm not yet fully sure this will bring enough benefits as there are not too many epoch infos "in flight" on the client side. So maybe I will simplify this part instead indeed, need to consider this

Reply:
Let's leave callbacks for now, b/c basically it would be nice to get rid of `EpochInfo` with its limit completely